src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,256 @@
+            {
+                Trace.Error($"The autologon user '{logonAccount}' doesnt have a user profile on the machine. Please login once with the expected autologon user and reconfigure the agent again");
+                throw new InvalidOperationException(StringUtil.Loc("NoUserProfile", logonAccount));
+            }
+
+            DisplayWarningsIfAny(regHelper);        
+            UpdateRegistriesForAutoLogon(regHelper, userName, domainName, logonPassword);
+            ConfigurePowerOptions();
+        }
+
+        public bool RestartNeeded()
+        {
+            var regHelper = new AutoLogonRegistryManager(HostContext.GetService<IWindowsRegistryManager>());            
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if (string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException(StringUtil.Loc("AutoLogonNotConfigured"));
+            }
+
+            return !_windowsServiceHelper.HasActiveSession(domainName, userName);
+        }
+
+        public void Unconfigure()
+        {
+            if(!_windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                Trace.Error("Needs Administrator privileges to unconfigure an agent running with AutoLogon capability.");          
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonRemoval"));
+            }
+            
+            /* We need to find out first if the AutoLogon was configured for the same user
+            If it is different user we should be reverting the AutoLogon user specific registries
+            */
+            var regManager = HostContext.GetService<IWindowsRegistryManager>();
+            var regHelper = new AutoLogonRegistryManager(regManager);
+            regHelper.FetchAutoLogonUserDetails(out string userName, out string domainName);
+            if (string.IsNullOrEmpty(userName) || string.IsNullOrEmpty(domainName))
+            {
+                throw new InvalidOperationException(StringUtil.Loc("AutoLogonNotConfigured"));
+            }
+
+            if (_windowsServiceHelper.HasActiveSession(domainName, userName))
+            {
+                Trace.Info($"AutoLogon is enabled for the same user, reverting the registry settings now.");
+                regHelper.RevertOriginalRegistrySettings();
+            }
+            else
+            {
+                Trace.Info($"AutoLogon is enabled for the different user {domainName}\\{userName}, reverting the registry settings now.");
+                var securityIdForTheUser = _windowsServiceHelper.GetSecurityId(domainName, userName);
+                AutoLogonRegistryManager regHelperForDiffUser = new AutoLogonRegistryManager(regManager, securityIdForTheUser);
+                regHelperForDiffUser.RevertOriginalRegistrySettings();
+            }            
+        }
+
+        public bool IsAutoLogonConfigured()
```

Comment:
What is the intent behind #1? As explained day before yesterday, we are dealing with many settings other than autologon e.g., startup process, screen saver, lock screen text etc, do you want me to wrap each of them in separate *RegistryManager(s)? What will be the gain? Will there be multiple individual calls from ConfigurationManager.Configure() then or having a single call and inside that taking care of all such settings will be ok? If it is latter, I had earlier named it as InteractiveSessionConfigManager which was doing all the required changes. 

I'll do #2 and #3.